core/modules/queue.cc

```diff
@@ -101,6 +109,11 @@ std::string Queue::GetDesc() const {
 void Queue::ProcessBatch(bess::PacketBatch *batch) {
   int queued =
       llring_mp_enqueue_burst(queue_, (void **)batch->pkts(), batch->cnt());
+  if (backpressure_ && !overload_ && llring_count(queue_) > high_water_) {
+    SignalOverload();
+    overload_ = true;
```

Comment:
Might you have SignalOverload() know to set overload_ properly rather than doing it outside?